extensions/classifiers/python-program.tokenizer.ts

```diff
@@ -26,322 +26,336 @@
-    var pseudoextras = groupOfRegEx('\\\\\\r?\\n|\\Z', comment, triple);
-    var pseudotoken = whitespace + groupOfRegEx(
-      pseudoextras, num, funny, contStr, name);
-
-    // Regular Expression object.
-    var tokenprog = new RegExp(token);
-    var pseudoprog = new RegExp(pseudotoken);
-    var single3prog = new RegExp(single3);
-    var double3prog = new RegExp(double3);
-
-    var endprogs = {
-      "'": new RegExp(single), '"': new RegExp(doubleQuote),
-      "'''": single3prog, '"""': double3prog,
-      "r'''": single3prog, 'r"""': double3prog,
-      "u'''": single3prog, 'u"""': double3prog,
-      "ur'''": single3prog, 'ur"""': double3prog,
-      "R'''": single3prog, 'R"""': double3prog,
-      "U'''": single3prog, 'U"""': double3prog,
-      "uR'''": single3prog, 'uR"""': double3prog,
-      "Ur'''": single3prog, 'Ur"""': double3prog,
-      "UR'''": single3prog, 'UR"""': double3prog,
-      "b'''": single3prog, 'b"""': double3prog,
-      "br'''": single3prog, 'br"""': double3prog,
-      "B'''": single3prog, 'B"""': double3prog,
-      "bR'''": single3prog, 'bR"""': double3prog,
-      "Br'''": single3prog, 'Br"""': double3prog,
-      "BR'''": single3prog, 'BR"""': double3prog,
-      r: null, R: null, u: null, U: null,
-      b: null, B: null
-    };
-
-    var tripleQuoted = [
-      "'''", '"""', "r'''", 'r"""', "R'''", 'R"""',
-      "u'''", 'u"""', "U'''", 'U"""', "ur'''", 'ur"""', "Ur'''", 'Ur"""',
-      "uR'''", 'uR"""', "UR'''", 'UR"""', "b'''", 'b"""', "B'''", 'B"""',
-      "br'''", 'br"""', "Br'''", 'Br"""', "bR'''", 'bR"""', "BR'''", 'BR"""'];
-
-    var singleQuoted = [
-      "'", '"', "r'", 'r"', "R'", 'R"', "u'", 'u"', "U'", 'U"', "ur'",
-      'ur"', "Ur'", 'Ur"', "uR'", 'uR"', "UR'", 'UR"', "b'", 'b"', "B'", 'B"',
-      "br'", 'br"', "Br'", 'Br"', "bR'", 'bR"', "BR'", 'BR"'];
-
-    var tabsize = 8;
-
-    return {
-      generateTokens: function(program) {
-        var tokenizedProgram = [];
-        var lnum = 0;
-        var parenlev = 0;
-        var continued = 0;
-        var namechars = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ_';
-        var numchars = '0123456789';
-        var contstr = '';
-        var needcont = 0;
-        var contline = null;
-        var indents = [0];
-        var lcount = 0;
-        var endprog = null;
-
-        while (1) {
-          var line = program[lcount];
-          lcount++;
-          if (line === undefined) {
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { LoggerService } from 'services/contextual/logger.service';
+import { ClassifiersExtensionConstants } from
+  'classifiers/classifiers-extension.constants.ts';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PythonProgramTokenizer {
+  private PythonProgramTokenType = (
+    ClassifiersExtensionConstants.PythonProgramTokenType);
+  constructor(private loggerService: LoggerService) {
```

Comment:
Maybe making this   constructor(private loggerService: LoggerService) {} will be better since it's an empty constructor, will be easier to read